### PR TITLE
Template index.html title

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -7,21 +7,55 @@
...
+
 export default defineConfig(({ mode }): UserConfig => {
   const commonEnv = loadEnv(mode, process.cwd(), "");
 
+  const themePath = fs.existsSync(external)
+    ? path.resolve(__dirname, external, "dist", "index.js")
+    : path.resolve(__dirname, internal, "index.tsx");
+  const file = fs.readFileSync(themePath, { encoding: "utf8" });
+  const match = /appNameFull: (?:"|')(.*)(?:"|'),*/.exec(file);
+  const appNameFull = match[1];
```
Comment (microbit-robert) [inline vite.config.ts:48]: Open to alternative, perhaps less fragile solutions. I'm not particularly fond of catering for both single and double quotes.

Comment (microbit-matt-hillsdon) [reply vite.config.ts:48]: Can we move the text we need here to a JSON file that index.js imports and we read directly?

**Comment by microbit-matt-hillsdon** [comment]
Parking this one for a bit to ponder a better approach.

File: src/components/DefaultPageLayout.tsx [hunk 2]
```diff
@@ -69,13 +68,6 @@ const DefaultPageLayout = ({
   const isSaveDialogClosed = useStore((s) => s.save.step === SaveStep.None);
 
   const toast = useToast();
-  const { appNameFull } = useDeployment();
-
-  useEffect(() => {
-    document.title = titleId
-      ? `${intl.formatMessage({ id: titleId })} | ${appNameFull}`
-      : appNameFull;
-  }, [appNameFull, intl, titleId]);
```
Comment (microbit-matt-hillsdon) [inline src/components/DefaultPageLayout.tsx:78]: If we pick this PR up again, then we need this or something equivalent back or the page title doesn't change when you navigate. The text for the page title itself is translated.

File: vite.config.ts (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply vite.config.ts:48]: No, because ESM.

Comment (microbit-matt-hillsdon) [reply vite.config.ts:48]: Alternative approach on https://github.com/microbit-foundation/ml-trainer/pull/402

**Comment by microbit-matt-hillsdon** [comment]
Closed in favour of merged https://github.com/microbit-foundation/ml-trainer/pull/402 which was based on this but async imports the external theme.

**Pull request closed without merging**